pkg/kgo/hooks.go

```diff
@@ -356,3 +362,43 @@ type HookFetchRecordUnbuffered interface {
 	// returned from polling.
 	OnFetchRecordUnbuffered(r *Record, polled bool)
 }
+
+/////////////
+// HELPERS //
+/////////////
+
+// implementsAnyHook will check the incoming Hook for any Hook implementation
+func implementsAnyHook(h Hook) bool {
+	switch h.(type) {
+	case HookNewClient:
+		return true
+	case HookBrokerConnect:
+		return true
+	case HookBrokerDisconnect:
+		return true
+	case HookBrokerWrite:
+		return true
+	case HookBrokerRead:
+		return true
+	case HookBrokerE2E:
+		return true
+	case HookBrokerThrottle:
+		return true
+	case HookGroupManageError:
+		return true
+	case HookProduceBatchWritten:
+		return true
+	case HookFetchBatchRead:
+		return true
+	case HookProduceRecordBuffered:
+		return true
+	case HookProduceRecordUnbuffered:
+		return true
+	case HookFetchRecordBuffered:
+		return true
+	case HookFetchRecordUnbuffered:
+		return true
+	default:
+		return false
+	}
```

Comment:
This can be squashed a bit,
```
case HookNewClient,
  HookBrokerConnect,
  HookBrokerDisconnect,
  ..all other hooks
  HookFetchRecordUnbuffered:
    return true
}
return false
```